guides/k8s-microservices/src/main/paradox/index.md

```diff
@@ -0,0 +1,292 @@
+
+* JDK8+
+* [Maven](https://maven.apache.org/)
+* [Docker](https://www.docker.com/)
+* [Minikube](https://kubernetes.io/docs/getting-started-guides/minikube/) with connectivity to it
+  via the `kubectl` command line tool
+* A clone of the [Lagom Chirper repository](https://github.com/lagom/activator-lagom-java-chirper)
+
+### Our approach
+
+This guide will be covering the resources required to run a Lagom system on Kubernetes. This will detail the various resource 
+types and their purpose, as well as how they need to be declared.
+
+### About Chirper
+
+Chirper is a Lagom-based microservices system that aims to simulate a Twitter-like website. It's configured for 
+both Maven and SBT builds but this guide will be using the Maven example.
+
+Deploying to Kubernetes requires the use of Docker images for each service. 
+Chirper leverages [fabric8's docker-maven-plugin](https://dmp.fabric8.io/) to produce Docker images as part of
+its build process.
+
+Kubernetes deployments also require the creation of various resources. Inside the `deploy/k8s` directory you'll find 
+the resources that are required to deploy the system. They'll be covered in detail below.
+
+### Kubernetes Resources
+
+Since Chirper uses many advanced features like Akka clustering, service discovery, ingress routing and more, support on
+Kubernetes uses many different types of resources. Reference the following to discover what kinds of resources are 
+used and why they're necessary.
+
+##### Pod
+
+The basic unit of execution in Kubernetes is a [Pod](https://kubernetes.io/docs/concepts/workloads/pods/pod/). A pod
+contains one or more containers that are co-located and co-scheduled. While Chirper doesn't use Pods directly, they
+are used by other resource types.
+
+##### StatefulSet
+
+To bootstrap the Akka cluster, Chirper and this guide make use of 
+[StatefulSets](https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#objectives). Pods that 
+are part of a StatefulSet have a sticky, unique identity that helps bootstrap the Akka cluster. 
+To do this, the first node is used as seed node and referenced by environment variables. Chirper defines `StatefulSet`
+resources for each of its services: `friendservice`, `activityservice`, `chirpservice`, and `web`
+
+##### Service
+
+A [Service](https://kubernetes.io/docs/concepts/services-networking/service/) provides the means to expose TCP and
+UDP ports to other pods within the cluster.
+
+Chirper requires TCP ports to be exposed for Akka remoting. This is done using a `Service` declaration. The exposed 
+port will be used by `StatefulSet` declarations to bootstrap the Akka cluster.
+
+Chirper also exposes various HTTP endpoints from the microservices defined within the Chirper itself. The TCP port 
+for each service is also exposed using `Service` declaration. The name `http-lagom-api` is used as the name
+thereby allowing these endpoints to be looked up via Kubernetes DNS using DNS SRV lookup.
```

Comment:
Should we talk about Lagom having to transform its DNS SRV lookup from `._tcp` to `_.http` due to Kubernetes only supporting `TCP` or `UDP` port in `Service`?